interfaces/builtin/fwupd.go

```diff
@@ -243,14 +252,15 @@ func (iface *fwupdInterface) AppArmorConnectedPlug(spec *apparmor.Specification,
 func (iface *fwupdInterface) AppArmorPermanentSlot(spec *apparmor.Specification, slot *snap.SlotInfo) error {
 	spec.AddSnippet(fwupdPermanentSlotAppArmor)
```

Comment:
Assuming fwupd could be snap or distro package, I think this should be (like you did below for seccomp):
```
// Only apply slot snippet when running as application snap
// on classic, slot side can be system or application
if !implicitSystemPermanentSlot(slot) {
    spec.AddSnippet(fwupdPermanentSlotAppArmor)
}
```

Reply:
Done.  I wonder if it actually matters though, since nothing will ever generate a security policy for the `system` pseudo-snap.

Reply:
@jhenstridge I'm not sure I understand the comment there. Maybe I can help to clarify something.  We do generate some policy for concrete system snaps, either core or snapd snap.